components/experimental/src/dimension/provider/units.rs

```diff
@@ -9,10 +9,78 @@
 //!
 //! Read more about data providers: [`icu_provider`]
 
-
 use alloc::borrow::Cow;
 use icu_provider::prelude::*;
 use tinystr::UnvalidatedTinyAsciiStr;
-use zerovec::{VarZeroVec, ZeroMap};
+use zerovec::{VarZeroVec, ZeroMap2d};
+
+// TODO: implement the units provider
+// thoughts:
+//   use the category (e.g. length, temperature, etc.) as secondary key
+//   have four maps:
+//     - one for the long width patterns (e.g. "{0} meter")
+//     - one for the short width (e.g. "{0} m"?)
+//     - one for the narrow width (e.g. "{0} m")
+//     - one for the DisplayNames (e.g. "meter")
+
+#[icu_provider::data_struct(UnitsDisplayNameV1Marker = "units/displaynames@1")]
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::dimension::provider::units),
+)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[yoke(prove_covariance_manually)]
+pub struct UnitsDisplayNameV1<'data> {
+    // TODO: store the pattern in a SinglePattern.
+    // TODO: use `MeasureUnit` for the units key instead of strings.
+    /// Contains the long width patterns for the units.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub long_width: ZeroMap2d<'data, str, Count, str>,
+
+    // TODO: store the pattern in a SinglePattern.
+    // TODO: use `MeasureUnit` for the units key instead of strings.
+    /// Contains the short width patterns for the units.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub short_width: ZeroMap2d<'data, str, Count, str>,
 
+    // TODO: store the pattern in a SinglePattern.
+    // TODO: use `MeasureUnit` for the units key instead of strings.
+    /// Contains the narrow width patterns for the units.
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub narrow_width: ZeroMap2d<'data, str, Count, str>,
+}
 
+// TODO: revise this.
+/// A CLDR plural keyword, or the explicit value 1.
+/// See <https://www.unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules>.
+#[zerovec::make_ule(CountULE)]
+#[zerovec::derive(Debug)]
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(
+    feature = "datagen", 
+    derive(serde::Serialize, databake::Bake),
+    databake(path = icu_experimental::compactdecimal::provider)
+)]
+#[repr(u8)]
+pub enum Count {
```

Comment:
Discussion: is this how we implement the `Count` for the `PluralRules`? or there is a more standardized way ?

Reply:
@zbraniecki This is another instance of a PluralCategory-like enum. We need this one because it has explicit cases included. Is there a better option?